python/ray/tune/ray_trial_executor.py

```diff
@@ -229,16 +231,35 @@ def fetch_result(self, trial):
         return result
 
     def _commit_resources(self, resources):
+        committed = self._committed_resources
+        custom_resources = {
+            res: (committed.get(res) + resources.get_res_total(res))
+            for res in resources.custom_resources
+        }
+        for resource, val in committed.custom_resources.items():
+            custom_resources.setdefault(resource, val)
```

Comment:
This is kind of hard to read. Why not.
all_keys = set(resources.custom_resources.keys() + committed.custom_resources.keys())
custom_resources = { res: a.get(k, 0) + b.get(k, 0) for k in all_keys }
?